### PR TITLE
chore: fix example `package.json`

### PR DESCRIPTION
**Pull request by danielroe**
This is a preliminary test towards updating default module packaging in https://github.com/nuxt/starter.

### CODE REVIEW

**Comment by userquin** [comment]
`types.d.ts` and `types.d.mts` should use `./module.js` in both imports

File: test/build.spec.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ describe('module builder', () => {
     const types = await readFile(join(distDir, 'types.d.ts'), 'utf-8')
     expect(types).toMatchInlineSnapshot(`
       "
-      import type { ModuleOptions, ModuleHooks, RuntimeModuleHooks, ModuleRuntimeConfig, ModulePublicRuntimeConfig } from './module'
+      import type { ModuleOptions, ModuleHooks, RuntimeModuleHooks, ModuleRuntimeConfig, ModulePublicRuntimeConfig } from './module.js'
```
Comment (danielroe) [inline test/build.spec.ts]:
```suggestion
      import type { ModuleOptions, ModuleHooks, RuntimeModuleHooks, ModuleRuntimeConfig, ModulePublicRuntimeConfig } from './module'
```

File: src/commands/build.ts [hunk 2]
```diff
@@ -155,13 +155,13 @@ async function writeTypes (distDir: string, meta: ModuleMeta) {
...
   const dtsContents = `
-import type { ${moduleImports.join(', ')} } from './module'
+import type { ${moduleImports.join(', ')} } from './module.js'
 
 ${appShims.length ? `declare module '#app' {\n${appShims.join('\n')}\n}\n` : ''}
 ${schemaShims.length ? `declare module '@nuxt/schema' {\n${schemaShims.join('\n')}\n}\n` : ''}
 ${schemaShims.length ? `declare module 'nuxt/schema' {\n${schemaShims.join('\n')}\n}\n` : ''}
 
-export type { ${typeExports[0].names.join(', ')} } from './module'
+export type { ${typeExports[0].names.join(', ')} } from './module.js'
```
Comment (danielroe) [inline src/commands/build.ts]:
```suggestion
export type { ${typeExports[0].names.join(', ')} } from './module'
```

File: src/commands/build.ts [hunk 3]
```diff
@@ -155,13 +155,13 @@ async function writeTypes (distDir: string, meta: ModuleMeta) {
   }
 
   const dtsContents = `
-import type { ${moduleImports.join(', ')} } from './module'
+import type { ${moduleImports.join(', ')} } from './module.js'
```
Comment (danielroe) [inline src/commands/build.ts]:
```suggestion
import type { ${moduleImports.join(', ')} } from './module'
```

File: test/build.spec.ts [hunk 4]
```diff
@@ -68,7 +68,7 @@ describe('module builder', () => {
       }
 
 
-      export type { ModuleHooks, ModuleOptions, ModulePublicRuntimeConfig, ModuleRuntimeConfig, RuntimeModuleHooks, default } from './module'
+      export type { ModuleHooks, ModuleOptions, ModulePublicRuntimeConfig, ModuleRuntimeConfig, RuntimeModuleHooks, default } from './module.js'
```
Comment (danielroe) [inline test/build.spec.ts]:
```suggestion
      export type { ModuleHooks, ModuleOptions, ModulePublicRuntimeConfig, ModuleRuntimeConfig, RuntimeModuleHooks, default } from './module'
```

**Pull request merged**